src/pybind/mgr/dashboard/frontend/src/app/ceph/rgw/rgw-bucket-form/rgw-bucket-form.component.ts

```diff
@@ -65,7 +66,41 @@ export class RgwBucketFormComponent implements OnInit {
       versioning: [null],
       'mfa-delete': [null],
       'mfa-token-serial': [''],
-      'mfa-token-pin': ['']
+      'mfa-token-pin': [''],
+      lock_enabled: [{ value: false, disabled: this.editing }],
+      lock_mode: ['COMPLIANCE'],
+      lock_retention_period_days: [
+        0,
+        [
+          CdValidators.number(false),
+          CdValidators.custom('eitherDaysOrYears', () => {
+            return (
+              self.bucketForm &&
+              _.get(self.bucketForm.getRawValue(), 'lock_enabled') &&
+              ((self.bucketForm.getValue('lock_retention_period_days') > 0 &&
+                self.bucketForm.getValue('lock_retention_period_years') > 0) ||
+                (self.bucketForm.getValue('lock_retention_period_days') === 0 &&
+                  self.bucketForm.getValue('lock_retention_period_years') === 0))
+            );
+          })
+        ]
+      ],
+      lock_retention_period_years: [
```

Comment:
Both period inputs use the same configuration, therefore there is no need to define it twice I would suggest something like this:

```diff
diff --git a/src/pybind/mgr/dashboard/frontend/src/app/ceph/rgw/rgw-bucket-form/rgw-bucket-form.component.ts b/src/pybind/mgr/dashboard/frontend/src/app/ceph/rgw/rgw-bucket-form/rgw-bucket-form.component.ts
index abaf843359..bd1cfcbd05 100644
--- a/src/pybind/mgr/dashboard/frontend/src/app/ceph/rgw/rgw-bucket-form/rgw-bucket-form.component.ts
+++ b/src/pybind/mgr/dashboard/frontend/src/app/ceph/rgw/rgw-bucket-form/rgw-bucket-form.component.ts
@@ -58,6 +58,15 @@ export class RgwBucketFormComponent implements OnInit {
 
   createForm() {
     const self = this;
+    const eitherDaysOrYears = CdValidators.custom('eitherDaysOrYears', () => {
+      if (!self.bucketForm || !_.get(self.bucketForm.getRawValue(), 'lock_enabled')) {
+        return false;
+      }
+      const years = self.bucketForm.getValue('lock_retention_period_years');
+      const days = self.bucketForm.getValue('lock_retention_period_days');
+      return (days > 0 && years > 0) || (days === 0 && years === 0);
+    });
+    const lockPeriodDefinition = [0, [CdValidators.number(false), eitherDaysOrYears]];
     this.bucketForm = this.formBuilder.group({
       id: [null],
       bid: [null, [Validators.required], this.editing ? [] : [this.bucketNameValidator()]],
@@ -69,38 +78,8 @@ export class RgwBucketFormComponent implements OnInit {
       'mfa-token-pin': [''],
       lock_enabled: [{ value: false, disabled: this.editing }],
       lock_mode: ['COMPLIANCE'],
-      lock_retention_period_days: [
-        0,
-        [
-          CdValidators.number(false),
-          CdValidators.custom('eitherDaysOrYears', () => {
-            return (
-              self.bucketForm &&
-              _.get(self.bucketForm.getRawValue(), 'lock_enabled') &&
-              ((self.bucketForm.getValue('lock_retention_period_days') > 0 &&
-                self.bucketForm.getValue('lock_retention_period_years') > 0) ||
-                (self.bucketForm.getValue('lock_retention_period_days') === 0 &&
-                  self.bucketForm.getValue('lock_retention_period_years') === 0))
-            );
-          })
-        ]
-      ],
-      lock_retention_period_years: [
-        0,
-        [
-          CdValidators.number(false),
-          CdValidators.custom('eitherDaysOrYears', () => {
-            return (
-              self.bucketForm &&
-              _.get(self.bucketForm.getRawValue(), 'lock_enabled') &&
-              ((self.bucketForm.getValue('lock_retention_period_days') > 0 &&
-                self.bucketForm.getValue('lock_retention_period_years') > 0) ||
-                (self.bucketForm.getValue('lock_retention_period_days') === 0 &&
-                  self.bucketForm.getValue('lock_retention_period_years') === 0))
-            );
-          })
-        ]
-      ]
+      lock_retention_period_days: lockPeriodDefinition,
+      lock_retention_period_years: lockPeriodDefinition
     });
   }
 
```

Reply:
Thx